core/storage/btree.rs

```diff
@@ -294,8 +341,16 @@ impl BTreeCursor {
                         _left_child_page,
                         _rowid,
                     }) => {
-                        if key < *_rowid {
-                            mem_page.advance();
+                        let SeekKey::TableRowId(rowid_key) = key else {
+                            unreachable!("table seek key should be a rowid");
+                        };
+                        mem_page.advance();
+                        let comparison = match cmp {
+                            SeekOp::GT => rowid_key <= *_rowid,
+                            SeekOp::GE => rowid_key < *_rowid,
+                            SeekOp::EQ => rowid_key < *_rowid,
```

Comment:
Shouldn't this be == ?

Reply:
I think this one is correct because if we're trying to find a record in a leaf node where `key == rowid`, then we need to go to `left_child_page` when ´key < interior_cell_rowid´, right?

However I think the index traversal is wrong because IIRC the index interior cells also contain actual data entries

Reply:
Wait nvm, if interior rowid is `10` then also rowid `10` is in the left child page (i.e. `<=`). I remembered that wrong.

Reply:
To answer your question:

Assume the search key is: rowid 100, and we want to find exactly that leaf node

if internal node rowid is over 100, we need to go left. if its exactly 100, we need to go left. otherwise we need to go right. hence `<=`

Assume the search key is: rowid 100, and we want to find the first leaf node that is greater or equal to our search key.

if internal node rowid is over 100, we need to go left, because greater or equal keys are on the left. if its exactly 100, we need to go left, because a leaf with rowid 100 is on the left. otherwise we need to go right. hence `<=`

Assume the search key is: rowid 100, and we want to find the first leaf node that is GREATER than our search key.

if internal node rowid is over 100, we need to go left, because greater keys exist on the left. if its exactly 100 or less, we need to go right, because there are no leaves with rowid>100 on the left.